sktime/base/_poly.py

```diff
@@ -0,0 +1,300 @@
+#!/usr/bin/env python3 -u
+# -*- coding: utf-8 -*-
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+"""Implements polymorphic base class."""
+
+__author__ = ["fkiraly", "miraep8"]
+__all__ = ["BasePolymorph"]
+
+
+from sktime.base import BaseObject
+from sktime.registry._lookup import _check_estimator_types
+
+
+class BasePolymorph(BaseObject):
+    """Handles parameter management for estimators composed of named estimators.
+
+    Partly adapted from sklearn utils.metaestimator.py.
+    """
+
+    def __init__(self, estimator_type="base"):
+
+        self.estimator_type = estimator_type
+        baseclass = _check_estimator_types(estimator_type)[0]
```

Comment:
I still think copying attributes from the first class in the polymorphism may not be the way to go?  ie - I think it would be nice to have something like `BasePolymorph` work for multiple types of composed estimators, and by enforcing this estimator to look like the first estimator provided may cause some issues/make that goal harder to realize.

Reply:
there is just one class? if you mean the `[0]`, this is since `_check_estimator_type` always coerces to list, but in this case it always is a one-element list.

Reply:
Ah you are correct, I had read this a bit too quickly, and thought this code was trying to do something different than it is.